src/wasm-validator.h

```diff
@@ -46,13 +46,15 @@
 namespace wasm {
 
 // Print anything that can be streamed to an ostream
-template <typename T>
+  template <typename T,
+    typename std::enable_if<
+      !std::is_base_of<Expression, typename std::remove_pointer<T>::type>::value
```

Comment:
this is starting to feel rather complicated. perhaps we should rethink the printing code?

e.g. in llvm I believe they support printing of objects, not pointers, which maybe makes things simpler (so you need to do `cerr << *x` instead of just `x` etc.).?